helix-core/src/diagnostic.rs

```diff
@@ -39,6 +39,10 @@ pub enum DiagnosticTag {
 #[derive(Debug, Clone)]
 pub struct Diagnostic {
     pub range: Range,
+    // whether this diagnostic ends at the end of(or inside) a word
+    pub ends_at_word: bool,
+    pub starts_at_word: bool,
+    pub zero_width: bool,
```

Comment:
Originally I thought this could be trivially computed but if I'm understanding correctly, a diagnostic may come in as zero-width and then changes may cause `range.start` and `range.end` to shift, so `range.start == range.end` may not detect this case after changes, right?

Reply:
Exactly these are the properties of the diagnostic when sent from the server. The idea is that non zero width diagnostics are removed if they are empty after mapping. But at that point we don't now if they were *previously* empty